modules/assignment/src/main/java/com/intuit/wasabi/assignment/impl/AssignmentsImpl.java

```diff
@@ -866,11 +866,15 @@ protected Experiment getExperiment(Application.Name applicationName, Experiment.
         Experiment result = null;
 
         if (metadataCacheEnabled) {
-            List<Experiment> experiments = metadataCache.getExperimentsByAppName(applicationName);
-            for (Experiment exp : experiments) {
-                if (experimentLabel.equals(exp.getLabel())) {
-                    result = exp;
-                    break;
+            //First fetch experiment list sorted by priorities
```

Comment:
@bargenilesh Would it make sense to mention something re: the fact that prioritized experiments are never the deleted ones? Just to clarify why you need to "First fetch experiment list sorted by priorities"